.grit/patterns/js/es6_imports.md

```diff
@@ -12,26 +12,57 @@ tags: #js, #es6, #migration, #cjs, #commonjs
 engine marzano(0.1)
 language js
 
+function transformProps($imports) {
+    $import_list = [],
+    $imports <: some bubble($import_list) {
+        or {
+            shorthand_property_identifier_pattern() as $key where $import_list += $key,
+            pair_pattern($key, $value) where $import_list += `$key as $value`
+        }
+    },
+    return join(list = $import_list, separator = ", "),
+}
+
 or {
     `const $sentry = require('@sentry/node')` => `import * as $sentry from '@sentry/node'`,
     // see https://github.com/motdotla/dotenv#how-do-i-use-dotenv-with-import
     `require("dotenv").config($config)` => `import * as dotenv from 'dotenv';\ndotenv.config($config)`,
 
-    `const { $imports } = require($source)` where {
-        $import_list = [],
-        $imports <: some bubble($import_list) {
-            or {
-                shorthand_property_identifier_pattern() as $key where $import_list += $key,
-                pair_pattern($key, $value) where $import_list += `$key as $value`
+    `const { $id: { $imports } } = require($specifier)` => `import { $id } from $specifier\nconst { $imports } = $id`,
+    `const { $imports } = require($specifier)` where { $transformed = transformProps($imports) } => `import { $transformed } from $specifier`,
+    `const $id = require($specifier).default` => `import $id from $specifier`,
+    `const $id = require($specifier).$named` => `import { $id } from $specifier` where { $id <: $named},
+    `const $id = require($specifier).$key` => `import { $key as $id } from $specifier`,
+
+    `const $id = $requireCall` where {
+        $requireCall <: contains r"require\(([^)]+)\)\.(.+)"($specifier, $rest)
+    } => `import __$id from $specifier;\nconst $id = __$id.$rest`,
```

Comment:
Not sure I follow this one - what's the purpose / why is a regex needed?

Reply:
It ain't pretty, but to illustrate the use case I would've ended up with something like this: https://app.grit.io/studio?key=wi3ghNEdeIZPmlZHgBP4b, which is only limiting to the cases explicitly defined.

The goal is to match everything behind `require($specifier).`

Is there a better way?

Reply:
I think this works: https://app.grit.io/studio?key=wi3ghNEdeIZPmlZHgBP4b

We generally try to avoid regex because it's hard to account for all the edge cases in parsing and whitespace.

Reply:
Thanks! I've used your idea and blended it into the existing solution.